libcaf_core/caf/function_view.hpp

```diff
@@ -195,16 +193,21 @@ class function_view {
   }
 
   void assign(type x) {
-    if (impl_.unsafe() && !x.unsafe())
+    if (!impl_ && x)
       new_self(x);
-    if (!impl_.unsafe() && x.unsafe())
+    if (impl_ && !x)
       self_.~scoped_actor();
     impl_.swap(x);
   }
 
+  void cleanup() {
```

Comment:
I'm not a big fan of the name `cleanup`. It's not used anywhere else in CAF. Maybe `reset`?